### ISSUE TITLE
Client.js Cannot read property 'message' of null

### ISSUE THREAD

**Issue by bmrpatel** [Azure/azure-iot-sdk-node#39]
# Context
- **OS and version used:** Windows 10, Ubuntu 16
- **Node.js version:** 7.9
- **npm version:** 4.2
- **list of installed packages:** 
"azure-iot-device": "^1.1.11",
    "azure-iot-device-amqp": "^1.1.11",
    "azure-iot-device-mqtt": "^1.1.11"

# Description of the issue:
Getting following error from iot-hub sdk. Application is not trying to do anything at this time. So it must be connection check or something that api is doing.

```
Cannot read property 'message' of null
    at d:\Projects\iothub\node_modules\azure-iot-device\lib\client.js:670:87
    at MqttBase.getReceiver (d:\Projects\iothub\node_modules\azure-iot-device-mqtt\lib\mqtt_base.js:186:9)
    at Mqtt.getReceiver (d:\Projects\iothub\node_modules\azure-iot-device-mqtt\lib\mqtt.js:110:20)
    at Client._getCurrentReceiver (d:\Projects\iothub\node_modules\azure-iot-device\lib\client.js:660:25)
    at Client._connectMethodReceiver (d:\Projects\iothub\node_modules\azure-iot-device\lib\client.js:645:14)
    at constructor._onEnter (d:\Projects\iothub\node_modules\azure-iot-device\lib\client.js:184:39)
    at constructor.transition (d:\Projects\iothub\node_modules\machina\lib\machina.js:653:28)
    at constructor._.each.Fsm.(anonymous function) [as transition] (d:\Projects\iothub\node_modules\machina\lib\machina.js:466:63)
    at safeUpdateSasCallback (d:\Projects\iothub\node_modules\azure-iot-device\lib\client.js:233:44)
    at d:\Projects\iothub\node_modules\azure-iot-device\lib\client.js:262:45
```

**Comment by pierreca** [comment]
there's definitely a bug here as there is a case where `err` could be null or undefined and we'd try to access its `message` property. I'm more worried about why you end up in this condition than the `message` thing though. I'll look into it, thanks!

**Comment by pierreca** [comment]
we've released a new package (azure-iot-device@1.1.12) with a fix.
Closing the issue but feel free to reopen if you still see the issue (meaning the fix doesn't work). 

Thanks again for reporting this!
